src/main/java/appeng/hooks/TickHandler.java

```diff
@@ -135,16 +139,17 @@ public void addCallable(final IWorld w, final IWorldCallable<?> c) {
 
     public void addInit(final AEBaseTileEntity tile) {
         // for no there is no reason to care about this on the client...
-        if (Platform.isServer()) {
-            this.getRepo().tiles.add(tile);
+        if (!tile.getWorld().isRemote()) {
+            this.getRepo().addTile(tile);
         }
     }
 
     private HandlerRep getRepo() {
-        if (Platform.isServer()) {
-            return this.server;
+        if (Platform.isClient()) {
```

Comment:
Is every call to getRepo now guarded with a check for the world? If so, I think we can drop this check.

Reply:
Probably, but I would keep it as sanity check in case some addon hooks into it. Better have it crash then whatever else happens.